state-synchronizer/src/coordinator.rs

```diff
@@ -635,6 +634,18 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             return Ok(());
         }
 
+        // If the request's epoch is in the past, `target_li` will be set to the end-of-epoch LI for that epoch
+        let target_li =
+            self.choose_response_li(request.known_version, request.current_epoch, target_li)?;
+        // Only populate highest_li field if it is different from target_li
+        let highest_li = if target_li.ledger_info().version() < local_version
```

Comment:
this is where I was confused, isn't this just, or we don't impl Eq for LI?
```
  let highest_li = self.local_state.highest_local_li.clone();
  let highest_li = Some(highest_li).filter(|li| li != target_li);
```

Reply:
if `target_li` is an end-of-epoch for a past epoch, this won't work - highest_li should be none, but this will return `Some`

Reply:
ah I'm too bad at reviewing